hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/protocolPB/OzoneManagerRequestHandler.java

```diff
@@ -378,6 +387,30 @@ private InfoBucketResponse infoBucket(InfoBucketRequest request)
     return resp.build();
   }
 
+  @RequestFeatureValidator(
+      conditions = ValidationCondition.OLDER_CLIENT_REQUESTS,
```

Comment:
This will block the request for all old clients, but a few versions in the future, clients can be old but still have EC support.

I think we could get the client version to this method by having `RequestValidations#validateResponse` (and `RequestValidations#validateRequest` as well) take the context as a parameter which they can forward to the annotated methods, instead of all the annotated methods receiving the same pre-constructed context. `OzoneManagerProtocolServerSideTranslatorPB#submitRequest` should have enough information to construct this context when invoking these methods.

Let me know what you think of this approach, or if you have a better idea.

Reply:
According to my plans, this I would like to clear before cutting the 1.3.0 release, and I would like to see the whole framework implemented at least on OM and SCM side, but probably on the client side as well.
I noticed this discrepancy also, so I took a note for my plans that I would re-formalize and publish as a design plan in the near future.

I have two competing ideas in mind for this,
one is to add the version to the annotation to mark when a validator has to run, so then the classification of the validators can be done in a per version base as the innermost structure in the registry. (this will lengthen the chain of map of map of map of lists to map of map of map of map of lists though).
the other is to add checks into the validators based on the OMRequest->version field, which is already given to all validators

I think the second one is sufficient for quite a while, and we can introduce the first one later, the first one is something that would perform and work well on the long run.

Reply:
I am ok with handling this in a follow-up to unblock the EC merge. Let's file a Jira and mark it as a 1.3.0 blocker if we go that route.